internal/identity/manager/manager.go

```diff
@@ -450,21 +357,70 @@ func (mgr *Manager) deleteSession(ctx context.Context, userID, sessionID string)
 	record := res.GetRecord()
 	record.DeletedAt = timestamppb.Now()
 
-	_, err = client.Put(ctx, &databroker.PutRequest{
+	_, err = mgr.cfg.Load().dataBrokerClient.Put(ctx, &databroker.PutRequest{
 		Records: []*databroker.Record{record},
 	})
 	if err != nil {
-		log.Error(ctx).Err(err).
+		log.Ctx(ctx).Error().Err(err).
 			Str("session_id", sessionID).
 			Msg("failed to delete session")
 		return
 	}
 }
 
-// reset resets all the manager datastructures to their initial state
-func (mgr *Manager) reset() {
-	mgr.sessions = sessionCollection{BTree: btree.New(8)}
-	mgr.users = userCollection{BTree: btree.New(8)}
+func (mgr *Manager) updateSession(ctx context.Context, s *session.Session) {
+	log.Ctx(ctx).Debug().
+		Str("user_id", s.GetUserId()).
+		Str("session_id", s.GetId()).
+		Msg("updating session")
+
+	fm, err := fieldmaskpb.New(s, "oauth_token", "id_token", "claims")
+	if err != nil {
+		log.Error(ctx).Err(err).
+			Str("user_id", s.GetUserId()).
+			Str("session_id", s.GetId()).
+			Msg("failed to create fieldmask for session")
+		return
+	}
+
+	_, err = session.Patch(ctx, mgr.cfg.Load().dataBrokerClient, s, fm)
+	if err != nil {
+		log.Error(ctx).Err(err).
+			Str("user_id", s.GetUserId()).
+			Str("session_id", s.GetId()).
+			Msg("failed to patch updated session record")
+		return
+	}
+
+	mgr.mu.Lock()
+	mgr.dataStore.putSession(s)
+	if rss, ok := mgr.refreshSessionSchedulers[s.GetId()]; ok {
+		rss.Update(s)
+	}
+	mgr.mu.Unlock()
+}
+
+// updateUser updates the user in the databroker, the local data store, and resets the scheduler
```

Comment:
I think it could be helpful to capture the reset explanation here (feel free to edit if this isn't quite right).
```suggestion
// updateUser updates the user in the databroker, the local data store, and resets the scheduler.
// (Whenever we refresh a session, we also refresh the user info. By resetting the user info 
// scheduler here we can avoid refreshing user info more often than necessary.)
```